fayyozbobokulov/tasks
# feat: added db setup

Summary
This Merge Request introduces the following features to containerize the NestJS app, set up PostgreSQL with Docker Compose, and configure Knex.js for database interaction.

Changes
Added Docker Support:

Containerized the NestJS application by creating a Dockerfile.
Configured Docker to build and run the app efficiently in any environment.
Added Docker Compose Configuration:

Set up docker-compose.yml to orchestrate both the PostgreSQL database and the NestJS app.
Ensured seamless connectivity between the app and the PostgreSQL container.
Integrated Knex for Database Queries:

Configured Knex.js as the SQL query builder for PostgreSQL.
Added relevant configurations in the project for easy migration and query handling.
Updated Documentation (README):

Enhanced the README.md to provide clear instructions on how to build, run, and connect the NestJS app and the database.
Included steps to use Docker and Docker Compose, making it easy for users to get started.
How to Test
Build and Run the Application:

Run the following command to build and start the containers:
bash
Copy code
docker-compose up --build
The app will be accessible at http://localhost:3000 and PostgreSQL at localhost:5432.
Verify Database Connection:

Ensure the NestJS app is connected to PostgreSQL via Knex.js.
Run migrations and ensure queries are functioning correctly.
Review Updated Documentation:

Check the updated README.md for instructions and ensure all steps are clear and accurate.
Notes
This MR improves the developer experience by containerizing the application and simplifying the database setup through Docker Compose.
Knex.js provides a scalable solution for managing database interactions.
